src/interfaces/State.java

```diff
@@ -0,0 +1,104 @@
+package interfaces;
```

Comment:
Like I mentioned above, we shouldn't keep all the interfaces together. Once I have the base package name, package name will need to change.

Reply:
done